### PR TITLE
[MDB IGNORE] Allows map elements to be loaded from TGM format files

### PR DESCRIPTION
**Pull request by ghost**
[vault][system][qol][tested]
Should probably also allow loading normal maps into the game as map elements too, if one really wants to.
Also converts all vaults and etc to this format too.

### CODE REVIEW

**Comment by Eneocho** [comment]
>Should probably also allow loading normal maps into the game as map elements too, if one really wants to.

Does this mean the fabled NT station vs Syndie station bus is now possible? Speedmerge it, quick, before kurf notices.

File: code/modules/awaymissions/maploader/reader.dm [hunk 1]
```diff
@@ -81,18 +82,32 @@ var/list/map_dimension_cache = list()
 	//first let's map model keys (e.g "aa") to their contents (e.g /turf/space{variables})
 	///////////////////////////////////////////////////////////////////////////////////////
 	var/list/grid_models = list()
-	var/key_len = length(copytext(tfile,2,findtext(tfile,quote,2,0)))//the length of the model key (e.g "aa" or "aba")
+	//the length of the model key (e.g "aa" or "aba")
+	var/key_len = length(copytext(tfile, findtext(tfile, quote), findtext(tfile, quote, findtext(tfile, quote) + 1, 0))) - 1
```
Comment (DamianX) [inline code/modules/awaymissions/maploader/reader.dm]: you're repeating `findtext(tfile, quote)` twice. Save that to a temporary variable instead

**Review by DamianX — COMMENTED** [review]
I think the uses of replacetext and the implementation of flip_map_key_axis are doing unnecessary work but I'd have to review this more carefully to know for sure

**Comment by adacovsk** [comment]
Almost 600,000 lines of new code, what a beast

File: code/modules/awaymissions/maploader/reader.dm (hunk 1 above)
Comment (ghost) [reply code/modules/awaymissions/maploader/reader.dm]: Fixed

**Comment by ghost** [comment]
> I think the uses of replacetext and the implementation of flip_map_key_axis are doing unnecessary work but I'd have to review this more carefully to know for sure

Removed most of that anyways

**Comment by Kurfursten** [comment]
You tested this right?

**Comment by ghost** [comment]
> You tested this right?

Extensively, even world.log'd the grid models and keys to verify they show up the same for tgm and dmm during development.

**Pull request merged**